### ISSUE TITLE
Psalted 2.0 (Reboot)

### ISSUE THREAD

**Issue by varoot** [chrisjbaik/psalted#44]
Hi guys,

I wanted to do a complete makeover/reboot of this project. Just so that I have something productive to do on my free time.

## Goals for the reboot:
- Use JavaScript for both back-end and front-end
- As much isomorphic as possible
- Use proper popular Javascript front-end framework such as React, Angular, Ember, or Vue (jQueryUI is bad)
- Stick to Material Design as much as possible for UI
- Use ES6+ throughout (with Babel to compile to front-end)
- Utilize new(er) technologies such as offline browsing, progressive web app, WebSocket, etc.
- Code/structure should make it easy to deploy on popular infrastructure (e.g. Docker, EC2, Heroku)
- Potentially portable to mobile or desktop app?

## Possible new features to be implemented/experimented:
- UI for chord input
- Presentation mode

I'm not really asking for permission or asking any of you to join me but let me know if you're interested or if you have suggestions on the framework/stack.

**Comment by chrisjbaik** [comment]
@varoot I'm a little out of touch with the web dev game as of recently so what you're suggesting is probably totally cool.

One thing I might suggest is seeing if there's a way we can have an integration with [Planning Center Online](https://planning.center). In Ann Arbor, we've actually transitioned our band's songbook management to this, and in addition, it is by far the most widely-used app for songbooks by churches in the US. Don't know if you had any thoughts on this?

**Comment by varoot** [comment]
@chrisjbaik looking through their API documentation, I think we can definitely do integration, but I wonder what data they can provide for Psalted. Aren't the songbooks uploaded as PDFs there? Maybe the integration would be the other way around where you manage the songs on Psalted and it's synced to Planning Center Online.

Currently I'm looking into [Vue.js](https://vuejs.org/) based stack. [Nuxt.js](https://nuxtjs.org/) will be the backend and I'm trying to see if I can get [Vonic](https://github.com/wangdahoo/vonic/) working but their documentation is all in Chinese. lol

I'm in the stage of planning, setting up project structure, exploring technologies right now. It's still in my local machine but let me know if anyone is interested in collaborating.

**Comment by chrisjbaik** [comment]
@varoot no, actually, there are [ChordPro](https://www.chordpro.org/) lyrics/chords on PCO. If it could be a two-way sync somehow, that would be great (although a headache considering conflicting changes and the like).

The API docs for those are [here](https://planningcenter.github.io/api-docs/#arrangements). (I think it will appear under the `chord_chart` attribute, and the example isn't particularly illuminating.

I probably won't be of much help in collaboration in the near future given my time constraints right now, but I can also try to recruit and hunt down people if you'd like (but you might prefer working alone for now). Let me know!
